test/production/api-navigation.spec.ts

```diff
@@ -1,83 +1,96 @@
+        expect(page.url()).toContain('/api/kotlinx.coroutines/');
     });
 
     test('Click on "Serialization" button should open the related page', async ({ page }) => {
-        const serializationButton = page.getByText('Serialization').first();
+        const serializationButton = await hoverOverApiElement(page, 'Serialization');
         await expect(serializationButton).toBeVisible();
         await serializationButton.click();
-        await expect(page.url()).toContain('/api/kotlinx.serialization/');
+        expect(page.url()).toContain('/api/kotlinx.serialization/');
     });
 
     test('Click on "Kotlin I/O library" button should open the related page', async ({ page }) => {
-        const ioButton = page.getByText('Kotlin I/O library').first();
+        const ioButton = await hoverOverApiElement(page, 'Kotlin I/O library');
         await expect(ioButton).toBeVisible();
         await ioButton.click();
-        await expect(page.url()).toContain('/api/kotlinx-io/');
+        expect(page.url()).toContain('/api/kotlinx-io/');
     });
 
     test('Click on "Date and time" button should open the related page', async ({ page }) => {
-        const datetimeButton = page.getByText('Date and time').first();
+        const datetimeButton = await hoverOverApiElement(page, 'Date and time');
         await expect(datetimeButton).toBeVisible();
         await datetimeButton.click();
-        await expect(page.url()).toContain('/api/kotlinx-datetime/');
+        expect(page.url()).toContain('/api/kotlinx-datetime/');
     });
 
-
     test('Click on "JVM Metadata" button should open the related page', async ({ page }) => {
-        const metadataButton = page.getByText('JVM Metadata').first();
+        const metadataButton = await hoverOverApiElement(page, 'JVM Metadata');
         await expect(metadataButton).toBeVisible();
         await metadataButton.click();
-        await expect(page.url()).toContain('/api/kotlinx-metadata-jvm/');
+        expect(page.url()).toContain('/api/kotlinx-metadata-jvm/');
     });
 
     test('Click on "Kotlin Gradle plugins" button should open the related page', async ({ page }) => {
-        const kgpButton = page.getByText('Kotlin Gradle plugins').first();
+        const kgpButton = await hoverOverApiElement(page, 'Kotlin Gradle plugins');
         await expect(kgpButton).toBeVisible();
         await kgpButton.click();
-        await expect(page.url()).toContain('/api/kotlin-gradle-plugin/');
+        expect(page.url()).toContain('/api/kotlin-gradle-plugin/');
     });
 
     test('Click on "Ktor" button should open the related page', async ({ page }) => {
-        const ktorButton = page.getByText('Ktor').first();
+        const ktorButton = await hoverOverApiElement(page, 'Ktor');
         await expect(ktorButton).toBeVisible();
         await ktorButton.click();
-        await expect(page.url()).toContain('api.ktor.io');
+        expect(page.url()).toContain('api.ktor.io');
```

Comment:
Is there a chance that the failure was caused by race condition between navigation and check?
What do you think about using something like `await page.waitForURL('https://api.ktor.io/');` here?

Reply:
Thanks, I've checked the click docs, seems valid.

There is another thought, though: it looks like `await expect(page).toHaveURL((/api\.ktor\.io/);
` is the better option because it will retry the check https://playwright.dev/docs/api/class-pageassertions#page-assertions-to-have-url
Looks like it is a better approach in general rather than checking page.url() which is instant and possibly could lead to flakiness